### PR TITLE
Add a lifecycle.Managed for AutoClosable's instances

### PR DESCRIPTION
**Pull request by frojasg**
I used ExecutorServiceManager as a guide.

### CODE REVIEW

**Comment by evnm** [comment]
Thanks for the contribution, @frojasg. This seems useful.

Could you add a JavaDoc to `AutoCloseableManager` providing some context, please?

**Comment by frojasg** [comment]
@evnm Thanks for the feedback. 

I added some documentation. Please let me know if I need to improve it 😄

File: dropwizard-lifecycle/src/main/java/io/dropwizard/lifecycle/AutoCloseableManager.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package io.dropwizard.lifecycle;
+
+/**
+ * An implementation of the Managed Interface for {@link AutoCloseable} instances. Adding an
+ * AutoCloseableManager instance to the application's environment ties that object’s lifecycle to
```
Comment (evnm) [inline dropwizard-lifecycle/src/main/java/io/dropwizard/lifecycle/AutoCloseableManager.java]: {@code AutoCloseableManager}

File: dropwizard-lifecycle/src/main/java/io/dropwizard/lifecycle/AutoCloseableManager.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    /**
+     * @param autoCloseable instance to close when the HTTP server stops.
+     */
+    public AutoCloseableManager(final AutoCloseable autoCloseable) {
+        this.autoCloseable = autoCloseable;
+    }
+
+    /**
+     * It does nothing
```
Comment (evnm) [inline dropwizard-lifecycle/src/main/java/io/dropwizard/lifecycle/AutoCloseableManager.java]: Let's phrase this "The start operation does nothing (i.e. it's a no-op)."

File: dropwizard-lifecycle/src/main/java/io/dropwizard/lifecycle/AutoCloseableManager.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+     */
+    @Override
+    public void start() throws Exception {
+    }
+
+    /**
+     * it calls {@link AutoCloseable#close()} given in the {@link AutoCloseableManager(AutoCloseable)}
```
Comment (evnm) [inline dropwizard-lifecycle/src/main/java/io/dropwizard/lifecycle/AutoCloseableManager.java]: "it calls" -> "Calls"

**Comment by frojasg** [comment]
Thank @evnm!

**Comment by evnm** [comment]
Thanks _you_ for contributing!

**Pull request merged**